ms2deepscore/data_generators.py

```diff
@@ -491,25 +491,92 @@ def _spectrum_pair_generator(self, batch_index: int) -> Iterator[SpectrumPair]:
             inchikey2 = self._find_match_in_range(inchikey1, target_score_range)
             spectrum1 = self._get_spectrum_with_inchikey(inchikey1)
             spectrum2 = self._get_spectrum_with_inchikey(inchikey2)
-            yield SpectrumPair(spectrum1, spectrum2)
+            score = self.reference_scores_df[inchikey1][inchikey2]
+            yield SpectrumPair(spectrum1, spectrum2, score)
+
     @ staticmethod
     def _data_selection(reference_scores_df, selected_inchikeys):
         """
         Select labeled data to generate from based on `selected_inchikeys`
         """
         return reference_scores_df.loc[selected_inchikeys, selected_inchikeys]
+
     def on_epoch_end(self):
         """Updates indexes after each epoch"""
         self.indexes = np.tile(np.arange(len(self.reference_scores_df)), int(self.settings["num_turns"]))
         if self.settings["shuffle"]:
             np.random.shuffle(self.indexes)
 
 
+class DataGeneratorCherrypicked(DataGeneratorBase):
```

Comment:
Should this still inherit from DataGeneratorBase. Or can the old DataGeneratorBase be removed or merged with this class? If the DataGeneratorBase is kept we should add the __init__  call to the superclass right?

Reply:
To me that was more of an interim solution. In the long run we might better switch to only one data generator. Well, maybe even in the short run, too. Because it is in principle up to us to decide what's a good generator for MS2DeepScore, the "older" ones are still available through older versions of the package. Keeping all three seems not very wise also from the maintenance point of view.